### PR TITLE
Removes use of definition I_CAN_LIVE_WITH_LNK4049 in e_os.h

### LINKED ISSUE #9332: Remove I_CAN_LIVE_WITH_LNK4049
In e_os.h there is the following:

```
#    elif _MSC_VER<1300 && defined(I_CAN_LIVE_WITH_LNK4049)
```

Looking at https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B, it appears that this is for MSVC *older than* Visual Studio .NET 2002 version 7.0

I think this renaming of stdin/out/err and such should be removed because of the age, and because this is a completely undocumented compile flag.  (A web search shows it only appearing in this file, as far as I can tell.)

### PR DESCRIPTION
**Pull request by frederikwh**
Fixes #9332

### CODE REVIEW

File: e_os.h [hunk 1]
```diff
@@ -156,22 +156,6 @@ FILE *__iob_func();
...
-          * or in other words with /MD. Declaring implicit import, i.e. with
-          * _imp_ prefix, works correctly with all compiler options, but
-          * without /MD results in LINK warning LNK4049: 'locally defined
-          * symbol "__iob" imported'.
-          */
-extern FILE *_imp___iob;
-#     define stdin  (&_imp___iob[0])
-#     define stdout (&_imp___iob[1])
-#     define stderr (&_imp___iob[2])
-#    endif
```
Comment (mattcaswell) [inline e_os.h]: This endif is still required

Comment (paulidale) [reply e_os.h]: I just noticed this too :(

Comment (frederikwh) [reply e_os.h]: I don’t know why it made sense to remove it. I will update accordingly.

**Comment by mattcaswell** [comment]
> There should be something in NOTES.WIN saying that we don't support MSVC older than whatever 1300 means.

This isn't actually removing support for that...only support for that in conjunction with this old undocumented macro.

**Comment by levitte** [comment]
That old undocumented macro is ours, is it not?  It's obviously there because the linker gets noisy, so element of least surprise led to a macro for users to basically tell the build that they know what they're doing.

Going back to the originating commit, e527201f6be3c295358bcc8b6bafec598f02dc97, this is kind of a special case (see the commit message), so as far as I understand, we're effectively removing the possibility to smoothly ilnk statically with msvcrt.lib for versions older than Visual Studio .NET 2002 version 7.0 (according to https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B)

**Comment by mattcaswell** [comment]
> we're effectively removing the possibility to smoothly ilnk statically with msvcrt.lib for versions older than Visual Studio .NET 2002 version 7.0 (according to https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B)

Right...but since that define was completely undocumented and buried away in the source code I suspect it is not widely used (or possibly not used at all). Rather that a NOTES.WIN entry, I think perhaps a CHANGES entry might be more appropriate.

**Comment by levitte** [comment]
So, er, it might be a good idea if someone finished this effort.

**Pull request closed without merging**